hal/src/core/pinmap_impl.h

```diff
@@ -1,64 +1,117 @@
-/**
- ******************************************************************************
- * @file    pinmap_impl.h
- * @author  Satish Nair, Matthew McGowan
- * @version V1.0.0
- * @date    25-Sept-2014
- * @brief   Implementation details of pins on STM32
- ******************************************************************************
-  Copyright (c) 2013-2015 Particle Industries, Inc.  All rights reserved.
-
-  This library is free software; you can redistribute it and/or
-  modify it under the terms of the GNU Lesser General Public
-  License as published by the Free Software Foundation, either
-  version 3 of the License, or (at your option) any later version.
-
-  This library is distributed in the hope that it will be useful,
-  but WITHOUT ANY WARRANTY; without even the implied warranty of
-  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-  Lesser General Public License for more details.
-
-  You should have received a copy of the GNU Lesser General Public
-  License along with this library; if not, see <http://www.gnu.org/licenses/>.
- ******************************************************************************
+/*
+ * Copyright (c) 2018 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
  */
 
-#ifndef PINMAP_IMPL_H
-#define	PINMAP_IMPL_H
+#pragma once
 
 #include "stm32f10x.h"
 
-
-#ifdef	__cplusplus
+#ifdef __cplusplus
 extern "C" {
 #endif
 
-typedef struct STM32_Pin_Info {
-  GPIO_TypeDef* gpio_peripheral;
-  pin_t gpio_pin;
-  uint8_t gpio_pin_source;
-  uint8_t adc_channel;
-  TIM_TypeDef* timer_peripheral;
-  uint16_t timer_ch;
-  PinMode pin_mode;
-  uint16_t timer_ccr;
-  int32_t user_property;
-} STM32_Pin_Info;
-
-/* Exported constants --------------------------------------------------------*/
+typedef struct Hal_Pin_Info {
+    GPIO_TypeDef    *gpio_peripheral;
+    pin_t           gpio_pin;
+    uint8_t         gpio_pin_source;
+    uint8_t         adc_channel;
+    TIM_TypeDef     *timer_peripheral;
+    uint16_t        timer_ch;
+    PinMode         pin_mode;
+    uint16_t        timer_ccr;
+    int32_t         user_property;
+} Hal_Pin_Info;
 
-extern STM32_Pin_Info PIN_MAP[];
-STM32_Pin_Info* HAL_Pin_Map(void);
+extern Hal_Pin_Info PIN_MAP[];
```

Comment:
Since we are already doing the refactoring, we should not expose this array directly and should convert any direct usage of it to use `HAL_Pin_Map()`. This applies to all the platforms.

Reply:
Removed.